neo/Network/P2P/Peer.cs

```diff
@@ -179,24 +183,35 @@ private void OnTcpConnected(IPEndPoint remote, IPEndPoint local)
             ImmutableInterlocked.Update(ref ConnectingPeers, p => p.Remove(remote));
             if (MaxConnections != -1 && ConnectedPeers.Count >= MaxConnections && !TrustedIpAddresses.Contains(remote.Address))
             {
-                Sender.Tell(Tcp.Abort.Instance);
+                Disconnect(DisconnectReason.MaxConnectionReached);
                 return;
             }
 
             ConnectedAddresses.TryGetValue(remote.Address, out int count);
             if (count >= MaxConnectionsPerAddress)
             {
-                Sender.Tell(Tcp.Abort.Instance);
+                Disconnect(DisconnectReason.MaxConnectionPerAddressReached);
+                return;
             }
             else
             {
                 ConnectedAddresses[remote.Address] = count + 1;
                 IActorRef connection = Context.ActorOf(ProtocolProps(Sender, remote, local), $"connection_{Guid.NewGuid()}");
                 Context.Watch(connection);
                 Sender.Tell(new Tcp.Register(connection));
-                ConnectedPeers.TryAdd(connection, remote);
+                ConnectedPeers.TryAdd(connection, new List<IPEndPoint> { remote });
             }
         }
+        private void Disconnect(DisconnectReason reason)
+        {
+            NetworkAddressWithTime[] networkAddresses = GetRandomConnectedPeers(AddrPayload.MaxCountToSend);
+            var payload = DisconnectPayload.Create(reason, networkAddresses.ToByteArray());
+            var disconnect = Message.Create(MessageCommand.Disconnect, payload);
+            var command = Tcp.Write.Create(ByteString.FromBytes(disconnect.ToArray()));
+
+            Sender.Tell(new Tcp.Register(Context.ActorOf(EmptyActor.Props())));
```

Comment:
Akka need a register to active the connection, otherwise, we can't send message between two sides.
The empty actor, which do nothing, as we just only want to send message, not care about the message recevied.